packages/venia-concept/src/ServiceWorker/Utilities/__tests__/htmlHandler.spec.js

```diff
@@ -0,0 +1,141 @@
+        this.headers = headers;
+    };
+    global.Response = function() {};
+});
+
+beforeEach(() => {
+    sendMessageToWindow.mockClear();
+});
+
+test('cacheHTMLPlugin should have cacheKeyWillBeUsed, requestWillFetch, fetchDidSucceed async functions implemented', () => {
+    matchFn.mockReturnValue(Promise.resolve(null));
+
+    expect(cacheHTMLPlugin).toHaveProperty('cacheKeyWillBeUsed');
+    expect(cacheHTMLPlugin).toHaveProperty('requestWillFetch');
+    expect(cacheHTMLPlugin).toHaveProperty('fetchDidSucceed');
+
+    expect(cacheHTMLPlugin.cacheKeyWillBeUsed).toBeInstanceOf(Function);
+    expect(cacheHTMLPlugin.requestWillFetch).toBeInstanceOf(Function);
+    expect(cacheHTMLPlugin.fetchDidSucceed).toBeInstanceOf(Function);
+
+    expect(cacheHTMLPlugin.cacheKeyWillBeUsed()).toBeInstanceOf(Promise);
+    expect(
+        cacheHTMLPlugin.requestWillFetch({
+            request: new Request('https://develop.pwa-venia.com/')
+        })
+    ).toBeInstanceOf(Promise);
+    expect(
+        cacheHTMLPlugin.fetchDidSucceed({
+            request: new Request('https://develop.pwa-venia.com/'),
+            response: new Response()
+        })
+    ).toBeInstanceOf(Promise);
+});
+
+test('cacheKeyWillBeUsed function should return a promise that resolves to /', async () => {
+    const response = await cacheHTMLPlugin.cacheKeyWillBeUsed();
+
+    expect(response).toBe('/');
+});
+
+test('requestWillFetch should return a new Request with url set to /', async () => {
+    const request = new Request('https://develop.pwa-venia.com/');
+    const response = await cacheHTMLPlugin.requestWillFetch({ request });
+
+    expect(response.url).not.toBe(request.url);
+});
+
+test('fetchDidSucceed should return the same response object given to it in params', async () => {
+    const request = new Request('https://develop.pwa-venia.com/');
+    const response = new Response();
+
+    const returnedResponse = await cacheHTMLPlugin.fetchDidSucceed({
+        request,
+        response
+    });
+
+    expect(returnedResponse).toBe(response);
+});
+
+test('fetchDidSucceed should not call sendMessageToWindow if the cache does not have a response for the url', async () => {
+    const request = new Request('https://develop.pwa-venia.com/');
+    const response = new Response();
+
+    await cacheHTMLPlugin.fetchDidSucceed({
+        request,
+        response
+    });
+
+    expect(sendMessageToWindow).not.toHaveBeenCalled();
+});
+
+test('fetchDidSucceed should call sendMessageToWindow if the response is different from what is in the cache', async () => {
+    /**
+     * Mocking Request to return different value when .text is called.
+     */
+    global.Response.prototype.text = function() {
+        return Promise.resolve(
+            `Hey I am random text ${Math.random()
+                .toString()
+                .slice(2)}`
+        );
+    };
+    global.Response.prototype.clone = function() {
+        return new Response();
+    };
```

Comment:
Don't forget to return these to their original states in case other tests rely on them.

Reply:
Nice catch. Done.